contracts/rewards/src/contract/execute.rs

```diff
@@ -231,12 +245,17 @@ where
 fn merge_rewards(
     rewards_1: HashMap<Addr, Uint128>,
     rewards_2: HashMap<Addr, Uint128>,
-) -> HashMap<Addr, Uint128> {
+) -> Result<HashMap<Addr, Uint128>, ContractError> {
     rewards_2
         .into_iter()
-        .fold(rewards_1, |mut rewards, (addr, amt)| {
-            *rewards.entry(addr).or_default() += amt;
-            rewards
+        .try_fold(rewards_1, |mut rewards, (addr, amt)| {
+            let r = rewards.entry(addr).or_default();
+            *r = r
+                .checked_add(amt)
+                .map_err(Into::<ContractError>::into)
+                .map_err(Report::from)?;
+
+            Ok(rewards)
```

Comment:
```suggestion
            Ok(r)
```

Reply:
this not failing a unit test indicates that we are missing coverage. Can you ensure we properly check the rewards?

Reply:
`r` it's a mutable reference to the rewards entry, so it mutates rewards. This is why tests don't fail. I have changed the code to make it less confusing